src/controllers/announcement_controller.cr

```diff
@@ -114,4 +118,17 @@ class AnnouncementController < ApplicationController
   private def user_id_param
     User.find_by_login(params["user"]?).try(&.id)
   end
+
+  private def check_rate_limit!
+    return true unless rate_limit = max_anns_per_hour
+    if current_user!.last_hour_announcements >= rate_limit
+      return redirect_to "/"
+    end
+  end
+
+  private def limit_reached?
+    if (rate_limit = max_anns_per_hour) && signed_in?
```

Comment:
`signed_in?` is already checked in the view. I think this is not a responsibility of this method and can be removed, so just:

```crystal
if rate_limit = max_anns_per_hour
  ...
```

WDYT ?

Reply:
You're right, it's redundant, I'm removing it.